src/app/core/services/filters.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { DEFAULT_DROPDOWN_FILTER, DropdownFilter } from '../../shared/issue-tables/dropdownfilter';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+/**
+ * Responsible for centralising filters
+ * Filters are subscribed to and emitted from this service
+ */
+export class FiltersService {
+  public dropdownFilter$ = new BehaviorSubject<DropdownFilter>(DEFAULT_DROPDOWN_FILTER);
+
+  clearFilters(): void {
+    this.dropdownFilter$.next(DEFAULT_DROPDOWN_FILTER);
+  }
+
+  updateSelectedLabels(newSelectedLabels: string[]): void {
```

Comment:
Seems like there's duplication of the logic when updating dropdownFilter itself. For instance, `updateSelectedLabels` is in essence doing the same thing as `updateHiddenLabels` and even the other methods: to update the right property. Instead, maybe we could follow a pattern of composition using the `Partial<T>` API in TypeScript?

This could look something like this:

```ts
updateFilters(newFilters: Partial<DropdownFilter>): void {
  ...this.dropdownFilter$.value,
  ...newFilters,
}
```

Then after we update, we can check our rules for updating the status and type. 

Consumers would then simply pass some filters that they want to change in the components themselves.

Reply:
Agreed. This is a type I was not aware of, it is perfect in this situation.
I have made the appropriate changes.